packages/vitest/src/node/plugins/index.ts

```diff
@@ -146,6 +147,7 @@ export async function VitestPlugin(options: UserConfig = {}, ctx = new Vitest())
     MocksPlugin(),
     GlobalSetupPlugin(ctx),
     CSSEnablerPlugin(ctx),
+    InstrumenterPlugin(ctx),
```

Comment:
Can this plugin be excluded like `UIPlugin` if coverage is disabled?

Reply:
Sure, we can add check for the `options.coverage?.enabled` here.

Note that there is a TODO for this in the plugin itself. It would be returning `null` which gets filtered out here. 

```ts
export function InstrumenterPlugin(ctx: Vitest): VitePlugin | null {
  // // Skip coverage reporters which do not need code transforms, e.g. native v8
  // TODO: This would be great but ctx has not yet been initialized
  // if (typeof ctx.coverageReporter.onFileTransform !== 'function')
  //  return null
```

I'm not 100% sure that one can be achieved though since `ctx` has not yet been initialized at this point. That solution would be ideal but since we don't need the `InstrumenterPlugin` at all with `c8`.

Reply:
Looks like `options` can only be used for resolving CLI flags here. The configuration file has not yet been processed - at this point we cannot know whether coverage is enabled by the configuration file only.

We'll need to revert this change and always enable the plugin.